src/simulator.zig

```diff
@@ -459,6 +461,28 @@ pub const Simulator = struct {
         return false;
     }
 
+    /// The core contains at least a view-change quorum of replicas. But if one or more of those
+    /// replicas are in status=recovering_head (due to corruption or state sync), then that may be
+    /// insufficient.
```

Comment:
Hm, this is interesting. I think for `.recovering_head` after a crash this is a fair game --- the replica isn't considered fully started until it recovered its head (VRR paper says "interesting, recovery protocol requires F+2 replicas to communicate"). 

But for the state sync, this doesn't seem to be entirely well-justified. Replica can get into `.recovering_head` due to state sync in normal processing without crashing if the network is laggy. So, ideally, if some replica is in this condition due to state sync, we should recover all-right. 

I think  a scenario here is: 3 replicas lag, 3 replicas race ahead, and then their leader crashes. As the three laggards are in `.recovering_head`, the cluster will be stuck, although only one replica has really crashed. 

This feels somewhat realistic, tbh, but also can be tackled separately (perhaps a `.recovering_head` replica can participate in a view change with a completely empty DVC, which could be treated as a bunch of NACKs if it's checkpoint is too old?)